packages/contracts-sdk/src/lib/contracts-sdk.ts

```diff
@@ -2436,4 +2442,26 @@ https://developer.litprotocol.com/v3/sdk/wallets/auth-methods/#auth-method-scope
       // },
     },
   };
+
+  private callWithGasMargin = async (
+    contract: ethers.Contract,
+    method: string,
+    args: any[],
+    overrides: ethers.PayableOverrides & { from?: string } = {},
+    percentageIncrease: number = GAS_LIMIT_INCREASE_PERCENTAGE
+  ) => {
+    if (!overrides.gasLimit) {
+      const txData = await contract.populateTransaction[method](
+        ...args,
+        overrides
+      );
+      const gasEstimation = await contract.provider.estimateGas(txData);
+      const adjustedGasLimit = gasEstimation
+        .mul(100 + percentageIncrease)
+        .div(100);
+      overrides.gasLimit = adjustedGasLimit;
```

Comment:
This is kind of a :nit: based on how the function is _currently_ being used, but in general it would be preferable to avoid mutating arguments passed due to the law of unintended consequences/don't surprise me principal

Reply:
Good call. Fixed with an internal _overrides